src/SFML/Window/OSX/WindowImplCocoa.mm

```diff
@@ -437,6 +438,27 @@ void showMouseCursor()
 }
 
 
+////////////////////////////////////////////////////////////
+void WindowImplCocoa::setMinimumSize(const std::optional<Vector2u>& minimumSize)
+{
+    WindowImpl::setMinimumSize(minimumSize);
+    const AutoreleasePool pool;
+    const NSSize          size = minimumSize ? NSMakeSize(minimumSize->x, minimumSize->y) : NSMakeSize(0, 0);
+    [m_delegate setMinimumSize:size];
+}
+
+
+////////////////////////////////////////////////////////////
+void WindowImplCocoa::setMaximumSize(const std::optional<Vector2u>& maximumSize)
+{
+    WindowImpl::setMaximumSize(maximumSize);
+    const AutoreleasePool pool;
+    const NSSize          size = maximumSize ? NSMakeSize(maximumSize->x, maximumSize->y)
+                                             : NSMakeSize(std::numeric_limits<CGFloat>::max(), std::numeric_limits<CGFloat>::max());
```

Comment:
Why do we set a maximum size of numeric_limits when no maximumSize is provided?

Reply:
It's just what the Cocoa API expects. `FLT_MAX` is the default maximum size.

https://developer.apple.com/documentation/appkit/nswindow/1419595-maxsize